core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -655,6 +655,15 @@ object SparkSubmit {
           // scalastyle:on println
         }
         System.exit(CLASS_NOT_FOUND_EXIT_STATUS)
+      case e: NoClassDefFoundError =>
```

Comment:
I think this OK. (PS the exception isn't "unreadable" -- just "uninformative"). `NoClassDefFoundError` is lower-level problem; classes were compiled in the presence of some other classes that aren't available at runtime. It's "worse" than `ClassNotFoundException`. It probably makes sense, but is this really a problem of building *without* Hive support? I would kind of think it's the opposite: you built with Hive and then it's not available at runtime?